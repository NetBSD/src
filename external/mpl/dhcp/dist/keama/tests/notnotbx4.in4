# double not boolean expression

# empty configs are not accepted by Kea
default-lease-time 1800;

# use double not in a reductible match if
class "reductible" {
    match if not (not ((option host-name = "www.example.com") or
                       (option host-name = "www.example.org")));
}

# use not with !=
class "other" {
    match if not (option host-name != "www.example.com");
}
