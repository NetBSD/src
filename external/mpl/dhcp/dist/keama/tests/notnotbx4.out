{
  # double not boolean expression
  # empty configs are not accepted by Kea
  "Dhcp4": {
    "valid-lifetime": 1800,
    "client-classes": [
      # use double not in a reductible match if
      {
        "name": "reductible",
        /// from: match if not not ((option dhcp.host-name) = 'www.example.com') or ((option dhcp.host-name) = 'www.example.org')
        "test": "(option[12].hex == 'www.example.com') or (option[12].hex == 'www.example.org')"
      },
      # use not with !=
      {
        "name": "other",
        /// from: match if not (option dhcp.host-name) != 'www.example.com'
        "test": "option[12].hex == 'www.example.com'"
      }
    ]
  }
}
